posthog/tasks/status_report.py

```diff
@@ -41,42 +49,63 @@ def status_report(*, dry_run: bool = False) -> Dict[str, Any]:
-                (team.id, report["period"]["start_inclusive"], report["period"]["end_inclusive"]),
-            )
-            team_report["persons_count_active_in_period"] = cursor.fetchone()[0]
-            cursor.execute(
-                sql.SQL(
-                    """
-                SELECT properties->>'$lib' as lib, COUNT(*) as count
-                FROM posthog_event WHERE team_id = %s AND created_at >= %s AND created_at <= %s GROUP BY lib
-            """
-                ),
-                (team.id, report["period"]["start_inclusive"], report["period"]["end_inclusive"]),
-            )
-            team_report["events_count_by_lib"] = {result.lib: result.count for result in namedtuplefetchall(cursor)}
-            cursor.execute(
-                sql.SQL(
-                    """
-                SELECT event as name, COUNT(*) as count
-                FROM posthog_event WHERE team_id = %s AND created_at >= %s AND created_at <= %s GROUP BY name
-            """
-                ),
-                (team.id, report["period"]["start_inclusive"], report["period"]["end_inclusive"]),
-            )
-            team_report["events_count_by_name"] = {result.name: result.count for result in namedtuplefetchall(cursor)}
+        params = (team.id, report["period"]["start_inclusive"], report["period"]["end_inclusive"])
+
+        team_report["persons_count_active_in_period"] = fetch_persons_count_active_in_period(params)
+        team_report["events_count_by_lib"] = fetch_event_counts_by_lib(params)
+        team_report["events_count_by_name"] = fetch_events_count_by_name(params)
+
         report["teams"][team.id] = team_report
+
     if not dry_run:
         posthoganalytics.api_key = "sTMFPsFhdP1Ssg"
         disabled = posthoganalytics.disabled
         posthoganalytics.disabled = False
         posthoganalytics.capture(get_machine_id(), "instance status report", report)
         posthoganalytics.disabled = disabled
     return report
+
+
+def fetch_persons_count_active_in_period(params: Tuple[Any, ...]) -> int:
+    return fetch_sql(
+        """
+        SELECT COUNT(DISTINCT person_id) as persons_count
+        FROM posthog_event JOIN posthog_persondistinctid ON (posthog_event.distinct_id = posthog_persondistinctid.distinct_id)
+        WHERE posthog_event.team_id = %s AND posthog_event.timestamp >= %s AND posthog_event.timestamp <= %s
+        """,
+        params,
+    )[0].persons_count
+
+
+def fetch_event_counts_by_lib(params: Tuple[Any, ...]) -> dict:
+    results = fetch_sql(
+        """
+        SELECT properties->>'$lib' as lib, COUNT(*) as count
+        FROM posthog_event WHERE team_id = %s AND timestamp >= %s AND timestamp <= %s
+        GROUP BY lib
+        """,
+        params,
+    )
+    return {result.lib: result.count for result in results}
+
+
+def fetch_events_count_by_name(params: Tuple[Any, ...]) -> dict:
+    results = fetch_sql(
+        """
+        SELECT event as name, COUNT(*) as count
```

Comment:
Not sure if it's related to this PR, but maybe we should obfuscate the name if the user has selected to anonymize their data?